src/components/bootcamps/layout/Layout.tsx

```diff
@@ -0,0 +1,287 @@
+                keepMounted
+                transformOrigin={{
+                  vertical: 'top',
+                  horizontal: 'right',
+                }}
+                open={Boolean(anchorElUser)}
+                onClose={handleCloseUserMenu}>
+                {settings.map((setting) => (
+                  <MenuItem key={setting} onClick={handleCloseUserMenu}>
+                    <Typography textAlign="center">{setting}</Typography>
+                  </MenuItem>
+                ))}
+              </Menu>
+            </Box>
+          </Toolbar>
+        </AppBar>
+        <Drawer variant="permanent" open={isDrawerOpen} onClose={toggleDrawer}>
+          <DrawerHeader>
+            <IconButton onClick={toggleDrawer}>
+              {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
+            </IconButton>
+          </DrawerHeader>
+          <Divider />
+          <List>
+            {/* {['Create', 'Starred', 'Send email', 'Drafts'].map((text, index) => (
+            <ListItem button key={text}>
+              <ListItemIcon>{index % 2 === 0 ? <AddBoxIcon /> : <MailIcon />}</ListItemIcon>
+              <ListItemText primary={text} />
+            </ListItem>
+          ))} */}
+            <ListItem>
+              <ListItemIcon>
+                <ListAltIcon></ListAltIcon>
+              </ListItemIcon>
+              <Button sx={{ justifyContent: 'space-between' }}>
+                <Link href="/bootcamps">All bootcamps</Link>
+              </Button>
+            </ListItem>
+            <ListItem>
+              <ListItemIcon>
+                <AddCircleIcon></AddCircleIcon>
+              </ListItemIcon>
+              <Button sx={{ justifyContent: 'space-between' }}>
+                <Link href="/bootcamps/create">Add bootcamp</Link>
+              </Button>
+            </ListItem>
+          </List>
+          <Divider />
+          <List>
+            <ListItem button onClick={handleOpenSettings}>
+              <ListItemIcon>
+                <Settings />
+              </ListItemIcon>
+              <ListItemText primary="Settings" />
+              {openCollapse ? <ExpandLess /> : <ExpandMore />}
+            </ListItem>
+            <Collapse in={openCollapse} timeout="auto" unmountOnExit>
+              <List component="div" disablePadding>
+                <ListItem button>
+                  <ListItemText inset primary="DropDown 1" />
+                </ListItem>
+                <ListItem button>
+                  <ListItemText inset primary="DropDown 2" />
+                </ListItem>
+                <ListItem button>
+                  <ListItemText inset primary="DropDown 3" />
+                </ListItem>
+              </List>
+            </Collapse>
+          </List>
+        </Drawer>
+        <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
+          <DrawerHeader />
+          <Typography paragraph></Typography>
+        </Box>
+      </Box>
+      <Footer></Footer>
```

Comment:
If the component requires no children you might as well write it as 
```suggestion
      <Footer />
```